internal/resourcepolicies/volume_conditions.go

```diff
@@ -0,0 +1,131 @@
+package resourcepolicies
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+type CSIVolumeSource struct {
+	Driver string `yaml:"driver"`
+}
+
+// VolumeConditions defined the current format of conditions we parsed
+type VolumeConditions struct {
+	Capacity     string           `yaml:"capacity,omitempty"`
+	StorageClass []string         `yaml:"storageClass,omitempty"`
+	NFS          *struct{}        `yaml:"nfs,omitempty"`
+	CSI          *CSIVolumeSource `yaml:"csi,omitempty"`
+}
+
+// Capacity consist of the lower and upper boundary
+type Capacity struct {
+	Lower resource.Quantity
+	Upper resource.Quantity
+}
+
+// ParseCapacity parse string into capacity format
+func ParseCapacity(capacity string) (*Capacity, error) {
+	capacities := strings.Split(capacity, ",")
+	var quantities []resource.Quantity
+	if len(capacities) != 2 {
+		return nil, fmt.Errorf("wrong format of Capacity %v", capacity)
+	} else {
+		for _, v := range capacities {
+			if quantity, err := resource.ParseQuantity(strings.TrimSpace(v)); err != nil {
+				return nil, fmt.Errorf("wrong format of Capacity %v with err %v", v, err)
+			} else {
+				quantities = append(quantities, quantity)
+			}
+		}
+	}
+	return &Capacity{Lower: quantities[0], Upper: quantities[1]}, nil
+}
+
+// IsInRanges returns true if the capacity range is in y, or it returns false
+func (c *Capacity) IsInRange(y *Capacity) bool {
+	return c.Lower.Cmp(y.Lower) >= 0 && c.Lower.Cmp(y.Lower) <= 0
+}
+
+// IsStorageClassMatch check target storage class is with in conditions storage class
+func IsStorageClassMatch(conditions *[]string, target string) bool {
+	target = strings.TrimSpace(target)
+	if len(*conditions) == 0 {
+		return true
+	}
+
+	for _, v := range *conditions {
+		sc := strings.TrimSpace(v)
+		if target == sc {
+			return true
+		}
+	}
+	return false
+}
+
+// IsConditionsMatch check current VolumeConditions matches current selector
+func (v *VolumeConditions) IsConditionsMatch(selector *VolumeConditions) bool {
```

Comment:
How about we introduce polymorphism at the condition level
```
type Condition interface {
// Check if the resource meet the condition
    Check(res) bool
// Validate if the condition which is parsed from the input yaml is valid
    Validate() 
}

```

 so there's `StorageClassCondition` `CapCondition`...  And these instances are created when you parse a single policy.
 
 There action may also that pattern in future b/c we want it to have parameters, and those need to be validated too, but that piece can be added later when needed.

Reply:
ok